src/Microsoft.Health.Fhir.TemplateManagement/Factory/StorageTemplateProviderFactory.cs

```diff
@@ -0,0 +1,49 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using Azure.Storage.Blobs;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Health.Fhir.TemplateManagement.ArtifactProviders;
+
+namespace Microsoft.Health.Fhir.TemplateManagement.Factory
+{
+    public class StorageTemplateProviderFactory : ITemplateProviderFactory
+    {
+        private BlobContainerClient _blobContainerClient;
+
+        private IMemoryCache _memoryCache;
+
+        private string _templateProviderCachePrefix = "storage-template-provider-";
+
+        public StorageTemplateProviderFactory(BlobContainerClient blobContainerClient, IMemoryCache memoryCache)
+        {
+            _memoryCache = memoryCache;
+            _blobContainerClient = blobContainerClient;
+        }
+
+        public IConvertDataTemplateProvider GetContainerRegistryTemplateProvider()
+        {
+            throw new NotImplementedException();
+        }
+
+        public IConvertDataTemplateProvider GetDefaultTemplateProvider()
+        {
```

Comment:
Just my 2c, since this factory class is named specific to StorageTemplate, on first glance, even though I know it is implementing ITemplateProviderFactory, it feels odd to see GetContainerRegistryTemplateProvider() and GetDefaultTemplateProvider() in here.
I think making this factory class more generic in interpretation then would help.
But if this is the pattern/practice for factory providers, then feel free to ignore.

Reply:
Looking at the existing code, the template provider contract is specific to the ACR implementation.  I think a larger refactor is order to align a common interface for consumers regardless of the source.

Reply:
Thanks for the feedback, agreed. I like the interface ITemplateProviderFactory just having a single method for GetTemplateProvider instead of the interface having methods for each template source. Then the factory can be injected based on options specified. There is still some cleanup required for some of the ACR template providers to not have ACR specific methods but they should be able to be switched to a generic ITemplateProviderFactory / GetTemplateProvider() method